slsim/roman_image_simulation.py

```diff
@@ -0,0 +1,315 @@
+    interp = InterpolatedImage(
+        Image(array, xmin=0, ymin=0),
+        scale=0.11 / oversample,
+        flux=np.sum(array),
+    )
+
+    # Gets psf and convolve
+    galsim_psf = get_psf(band, detector, detector_pos, oversample, psf_directory)
+    convolved = galsim.Convolve(interp, galsim_psf)
+
+    # Draw interpolated image at the original (not oversampled) pixel scale
+    im = galsim.ImageF(num_pix, num_pix, scale=0.11)
+    im.setOrigin(0, 0)
+    image = convolved.drawImage(im)
+
+    if add_noise:
+        # Obtain sky background corresponding to certain band and add it to the image
+        # Requires webbpsf data files to use
+        image = add_roman_background(
+            image, band, detector, num_pix, _exposure_time, ra, dec, date
+        )
+
+        # Add detector effects and get the resulting array
+        rng = galsim.UniformDeviate(seed)
+        roman.allDetectorEffects(
+            image, prev_exposures=(), rng=rng, exptime=_exposure_time
+        )
+
+    array = image.array
+
+    final_array = array[3:-3, 3:-3]
+    final_array = final_array / _exposure_time
+    return final_array
+
+
+# The following functions have been copy-pasted from the mejiro repo
+# Credit to Bryce Wedig
+
+
+def get_psf(band, detector, detector_pos, oversample, psf_directory):
+    """Obtain galsim psf corresponding to specific band, using webbpsf.
+
+    :param band: The specific band corresponding to the psf
+    :type band: string
+    :param detector: The specific Roman detector being used to generate the psf
+    :type detector: integer from 1 to 18
+    :param detector_pos: The position of the detector being used to generate the psf
+    :type detector_pos: integer between 4 + num_pix * oversample and 4092 - num_pix * oversample
+    :param oversample: Number of times that each pixel's side is subdivided for higher accuracy psf convolution
+    :type oversample: integer
+    :param psf_directory: Path to directory containing psf file(s) where the psf can be loaded.
+                            Otherwise, the psf will be generated by webbpsf which is very slow
+    :type psf_directory: string
+    :return: An image of the psf generated by webbpsf
+    :rtype: galsim's InterpolatedImage class
+    """
+    detector = f"SCA{str(detector).zfill(2)}"
+    # Since generating the webbpsf is very slow, it can alternatively be loaded from a pickle file
+    # where the psf has been generated ahead of time
+    psf_file_name = (
+        f"{band}_{detector}_{detector_pos[0]}_{detector_pos[1]}_{oversample}.pkl"
+    )
+    if psf_directory is not None:
+        psf_file_path = os.path.join(psf_directory, psf_file_name)
+    else:
+        psf_file_path = os.path.join(
+            os.path.dirname(__file__), "..", "data", "webbpsf", psf_file_name
+        )
+
+    if os.path.exists(psf_file_path):
+        with open(psf_file_path, "rb") as psf_file:
+            psf = pickle.load(psf_file)
+    else:
+        wfi = WFI()
```

Comment:
is it possible to test these lines of code?

Reply:
No, these lines of code are the ones that require external files from webbpsf to generate the psf